### ISSUE TITLE
HmIP-SMI55 - lost motion and illumination entities

### ISSUE THREAD

**Issue by angelnu** [danielperna84/pyhomematic#303]
**Home Assistant version (if applicable):**
0.109.3

**Describe the bug**
After adding the remote base class with 
https://github.com/danielperna84/pyhomematic/commit/b747e0f1f3df6c8366e23ef17533c7fb915ded7f the illumination and motion for HmIP-SMI55 do not longer get reported and display as 'restored' in home assistant. Reverting back to current master version address the issue but then no button events are reported.

I initially thought this was due to the diamond inheritance but this should not really be a problem with python 3.

Then I noticed that both Remote and MotionIPV2 define 'ELEMENT'. According to the inheritance then the method from the Remote class would be used. I am going to try to explicitly define ELEMENT' in the IPRemoteMotionV2 but to make sure I do not mess it I would appreciate some explanation for 'ELEMENT'

What is the 'ELEMENT' class support to report:
- channels that have the same type? [1,2]
- unique channels? [0,1,3]
- all channels? [0,1,2,3]

**To Reproduce**
1. Upgrade to pythomematic devel branch
2. illumination and motion not reported 

**Expected behavior**
All entities reported and button events working

**Screenshots**
<img width="792" alt="image" src="https://user-images.githubusercontent.com/4406403/80954642-11ea5180-8dfe-11ea-8a2d-94639ad37748.png">

**Additional context**
Regression introduced by https://github.com/danielperna84/pyhomematic/commit/b747e0f1f3df6c8366e23ef17533c7fb915ded7f

**Comment by angelnu** [comment]
Tested with:
```

class IPRemoteMotionV2(Remote, MotionIPV2):
    """Motion detection with buttons (hm ip).
       This is a binary sensor."""

    # pylint: disable=useless-super-delegation
    def __init__(self, device_description, proxy, resolveparamsets=False):
        super().__init__(device_description, proxy, resolveparamsets)
    
    @property
    def ELEMENT(self):
        return [0, 1, 3]
```

No success: illumination and motion stay unavailable.

**Comment by danielperna84** [comment]
I think the problem actually is, that I forgot to add the new class [here](https://github.com/home-assistant/core/blob/8e071b69f4c2d45f3a8df8e5783bd0cd029c605f/homeassistant/components/homematic/const.py#L42). Could you please test this? Additionally, it probably also has to get mentioned [here](https://github.com/home-assistant/core/blob/8e071b69f4c2d45f3a8df8e5783bd0cd029c605f/homeassistant/components/homematic/binary_sensor.py#L18) as well to be displayed as the correct type.

**Comment by angelnu** [comment]
Ok, thanks for giving me the pointers! - I will give it a try today/tomorrow and report back.

Could you please explain me then what the ELEMENT property is? I tried to figure it out (see above my speculation) but could not get a consistent result by just looking at the code...

**Comment by danielperna84** [comment]
It's a bit messy. Originally it was intended to provide the _main_ channels of a device. Those, for which entities would be created. Switches and others inherit methods to interact with the devices. These methods often point to `ELEMENT`. You can see this [here](https://github.com/danielperna84/pyhomematic/blob/91c91cbfb6b283f2a5815d33f3e7396535ad0f31/pyhomematic/devicetypes/helper.py#L171) for the `LEVEL` parameters of blinds, or further below for other functionality.
The problem with this approach is, that this only really works well for devices that don't have a lot of functionality. So often times you'll find code in the init-blocks of classes where the parameter-channel-mapping is done, along with methods which can be used for the device type.

So essentially this was meant to make things easier by allowing to reuse code, but it sort of breaks when there is too much functionality within a single device.

**Comment by angelnu** [comment]
I see, so when a device uses two helpers that relay on ELEMENT and they are not in the same channel the method does not work. Ok, I think I get it now...

Your recommend fix did work: https://github.com/angelnu/core/commit/93b2a20ccbaa578659d4fbb2c9b75f60eba5ee6f

I did not do a PR since you have not published 0.1.67 - let me know if you want me to do it once you are ready.

**Comment by danielperna84** [comment]
I have created a PR here for 0.1.67 an noted what needs to be done in HA once it is released. Therefore I'll close this issue. Thanks for figuring this out before the release. 👍
